### PR TITLE
Disable keepalive on Mac

### PR DESCRIPTION
**Pull request by rhc54**
The Mac appears to have problems with the keepalive support - once keepalive starts, the memory footprint soars. So disable keepalive on the Mac

(cherry picked from commit open-mpi/ompi@8e30579e6efab580cf9cf1bec8f8df1376b7e9ef)

@jsquyres Please review

### CODE REVIEW

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/277/

**Build Log**
_last 50 lines_

```
[...truncated 7450 lines...]
libtool: finish: PATH="/hpc/local/bin::/usr/local/bin:/bin:/usr/bin:/usr/sbin:/hpc/local/bin:/hpc/local/bin/:/hpc/local/bin/:/sbin:/usr/sbin:/bin:/usr/bin:/usr/local/sbin:/opt/ibutils/bin:/sbin" ldconfig -n /var/lib/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace/ompi_install1/lib/openmpi
----------------------------------------------------------------------
Libraries have been installed in:
   /var/lib/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace/ompi_install1/lib/openmpi

If you ever happen to want to link against installed libraries
in a given directory, LIBDIR, you must either use libtool, and
specify the full pathname of the library, or use the `-LLIBDIR'
flag during linking and do at least one of the following:
   - add LIBDIR to the `LD_LIBRARY_PATH' environment variable
     during execution
   - add LIBDIR to the `LD_RUN_PATH' environment variable
     during linking
   - use the `-Wl,-rpath -Wl,LIBDIR' linker flag
   - have your system administrator add LIBDIR to `/etc/ld.so.conf'

See any operating system documentation about shared libraries for
more information, such as the ld(1) and ld.so(8) manual pages.
----------------------------------------------------------------------
make[3]: Leaving directory `/scrap/jenkins/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace/orte/mca/odls/default'
make[2]: Leaving directory `/scrap/jenkins/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace/orte/mca/odls/default'
Making install in mca/oob/tcp
make[2]: Entering directory `/scrap/jenkins/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace/orte/mca/oob/tcp'
  CC       oob_tcp.lo
  CC       oob_tcp_component.lo
  CC       oob_tcp_listener.lo
  CC       oob_tcp_common.lo
  CC       oob_tcp_connection.lo
  CC       oob_tcp_sendrecv.lo
oob_tcp_component.c: In function 'tcp_component_register':
oob_tcp_component.c:432: error: 'mca_oob_tcp_component_t' has no member named 'retry_delay'
oob_tcp_component.c:438: error: 'mca_oob_tcp_component_t' has no member named 'retry_delay'
oob_tcp_component.c:440: error: 'mca_oob_tcp_component_t' has no member named 'max_recon_attempts'
make[2]: *** [oob_tcp_component.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory `/scrap/jenkins/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace/orte/mca/oob/tcp'
make[1]: *** [install-recursive] Error 1
make[1]: Leaving directory `/scrap/jenkins/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace/orte'
make: *** [install-recursive] Error 1
Build step 'Execute shell' marked build as failure
TAP Reports Processing: START
Looking for TAP results report in workspace using pattern: **/*.tap
Did not find any matching files.
Anchor chain: could not read file with links: /var/lib/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace/jenkins_sidelinks.txt (No such file or directory)
Setting commit status on GitHub for https://github.com/open-mpi/ompi-release/commit/6a083f24cadaa8450aa009585466848ca585c45d
[BFA] Scanning build for known causes...
[BFA] No failure causes found
[BFA] Done. 0s
Setting status of 93c16da1a645a44f1de4cbb9ae59a6b4d410c3ae to FAILURE with url http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr/277/ and message: Build finished.
```

Test FAILed.

File: orte/mca/oob/tcp/oob_tcp_component.c [hunk 1]
```diff
@@ -427,11 +427,20 @@ static int tcp_component_register(void)
                                           OPAL_INFO_LVL_9,
                                           MCA_BASE_VAR_SCOPE_READONLY,
                                           &mca_oob_tcp_component.keepalive_probes);
+#endif
     
-    mca_oob_tcp_component.skip_version_check = false;
-    (void)mca_base_component_var_register(component, "skip_version_check",
-                                          "Skip checking versions between connections",
-                                          MCA_BASE_VAR_TYPE_BOOL, NULL, 0, 0,
+    mca_oob_tcp_component.retry_delay = 0;
```
Comment (jsquyres) [inline orte/mca/oob/tcp/oob_tcp_component.c]: @rhc54 Looks like you pulled over some extra stuff in this PR; that's what Jenkins' compile failure is about.

**Comment by jsquyres** [comment]
FWIW, once the extra stuff is snipped out of this PR, it should probably be good.

Should probably cite @scrobey in the commit message.

I can confirm that running this on the v1.8 branch makes orterun not blow up in memory usage:

```
mpirun --mca oob_tcp_keepalive_time 0 sleep 1000
```

(i.e., this PR will effectively make this behavior be the default on OS X)

Others who care about this issue: @ggouaillardet @rolfv @ianhinder @barrywardell

**Comment by rhc54** [comment]
@jsquyres I removed the bad code - not sure how it crept in there, but it's gone now. Will wait for tests on revised version and a quick look by you before committing.

Thanks!

**Comment by mellanox-github** [comment]
- Coverity found 1467 errors for all: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/all_278/output/errors/index.html
- Coverity found 23 errors for oshmem: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/oshmem_278/output/errors/index.html
- Coverity found 7 errors for fca: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/fca_278/output/errors/index.html <br><br>:octocat: _Sent from [GH](http://nodegh.io)._

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/278/
Test PASSed.

**Comment by mellanox-github** [comment]
- Coverity found 1467 errors for all: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/all_279/output/errors/index.html
- Coverity found 23 errors for oshmem: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/oshmem_279/output/errors/index.html
- Coverity found 7 errors for fca: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/fca_279/output/errors/index.html <br><br>:octocat: _Sent from [GH](http://nodegh.io)._

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/279/

**Build Log**
_last 50 lines_

```
[...truncated 43623 lines...]
shmemcc -g oshmem_shmalloc.c -o oshmem_shmalloc
make[2]: Leaving directory `/scrap/jenkins/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace-2/ompi_install1/examples'
make[2]: Entering directory `/scrap/jenkins/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace-2/ompi_install1/examples'
shmemcc -g oshmem_circular_shift.c -o oshmem_circular_shift
make[2]: Leaving directory `/scrap/jenkins/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace-2/ompi_install1/examples'
make[2]: Entering directory `/scrap/jenkins/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace-2/ompi_install1/examples'
shmemcc -g oshmem_max_reduction.c -o oshmem_max_reduction
make[2]: Leaving directory `/scrap/jenkins/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace-2/ompi_install1/examples'
make[2]: Entering directory `/scrap/jenkins/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace-2/ompi_install1/examples'
shmemcc -g oshmem_strided_puts.c -o oshmem_strided_puts
make[2]: Leaving directory `/scrap/jenkins/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace-2/ompi_install1/examples'
make[2]: Entering directory `/scrap/jenkins/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace-2/ompi_install1/examples'
shmemcc -g oshmem_symmetric_data.c -o oshmem_symmetric_data
make[2]: Leaving directory `/scrap/jenkins/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace-2/ompi_install1/examples'
make[2]: Entering directory `/scrap/jenkins/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace-2/ompi_install1/examples'
shmemfort -g hello_oshmemfh.f90 -o hello_oshmemfh
make[2]: Leaving directory `/scrap/jenkins/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace-2/ompi_install1/examples'
make[2]: Entering directory `/scrap/jenkins/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace-2/ompi_install1/examples'
shmemfort -g ring_oshmemfh.f90 -o ring_oshmemfh
make[2]: Leaving directory `/scrap/jenkins/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace-2/ompi_install1/examples'
make[1]: Leaving directory `/scrap/jenkins/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace-2/ompi_install1/examples'
make: Leaving directory `/scrap/jenkins/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace-2/ompi_install1/examples'
+ for exe in hello_c ring_c
+ exe_path=/var/lib/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace-2/ompi_install1/examples/hello_c
+ PATH=/var/lib/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace-2/ompi_install1/bin:/hpc/local/bin::/usr/local/bin:/bin:/usr/bin:/usr/sbin:/hpc/local/bin:/hpc/local/bin/:/hpc/local/bin/:/sbin:/usr/sbin:/bin:/usr/bin:/usr/local/sbin:/opt/ibutils/bin
+ LD_LIBRARY_PATH=/var/lib/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace-2/ompi_install1/lib:
+ mpi_runner 8 /var/lib/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace-2/ompi_install1/examples/hello_c
+ local np=8
+ local exe_path=/var/lib/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace-2/ompi_install1/examples/hello_c
+ local exe_args=
+ local 'common_mca=-bind-to core'
+ local 'mca=-bind-to core'
+ '[' yes == yes ']'
+ timeout -s SIGKILL 10m mpirun -np 8 -bind-to core -mca pml ob1 -mca btl self,tcp /var/lib/jenkins/jobs/gh-ompi-release-v1.8-pr/workspace-2/ompi_install1/examples/hello_c
[jenkins01:01306] mca_oob_tcp_accept: accept() failed: Bad file descriptor (9).
./jenkins_scripts/jenkins/ompi/ompi_jenkins.sh: line 148:  1305 Killed                  $timeout_exe mpirun -np $np $mca -mca pml ob1 -mca btl self,tcp ${exe_path} ${exe_args}
Build step 'Execute shell' marked build as failure
TAP Reports Processing: START
Looking for TAP results report in workspace using pattern: **/*.tap
Saving reports...
TAP Reports Processing: FINISH
coverity_for_all    http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/all_279/output/errors/index.html
coverity_for_oshmem http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/oshmem_279/output/errors/index.html
coverity_for_fca    http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/fca_279/output/errors/index.html
Setting commit status on GitHub for https://github.com/open-mpi/ompi-release/commit/67e467096d63e2c0ad27f3e67e41d336c9d046de
[BFA] Scanning build for known causes...
[BFA] No failure causes found
[BFA] Done. 0s
Setting status of 3b24074fcd1e0253d11eaef78b3ddb883acf4775 to FAILURE with url http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr/279/ and message: Build finished.
```

Test FAILed.

**Comment by jsquyres** [comment]
Looks like this isn't quite ready for prime time yet (Jenkins fail, and George's suggestion to use **APPLE**).  I'm hopping on a plane shortly; I should be able to look at this when I get home later today.

**Comment by rhc54** [comment]
I won't really be able to look at it in detail until I get home mid-next week. Nothing urgent about it as we aren't about to release a 1.8.6. I have no real issue with using the APPLE flag, though I will note that we use OPAL_HAVE_SOLARIS for similar issues, also set in check_os_flavors. So I was only following precedent.

**Comment by rhc54** [comment]
Okay, I was able to squeeze some time in on odin and got it fixed. Also made George's requested change. See what you think.

**Comment by mellanox-github** [comment]
- Coverity found 1467 errors for all: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/all_281/output/errors/index.html
- Coverity found 23 errors for oshmem: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/oshmem_281/output/errors/index.html
- Coverity found 7 errors for fca: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/fca_281/output/errors/index.html <br><br>:octocat: _Sent from [GH](http://nodegh.io)._

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/281/
Test PASSed.

**Comment by rhc54** [comment]
I think we all agree that Jeff's solution is the one we will be using, so I'll withdraw this one.

**Pull request closed without merging**